ui/app/pages/confirm-approve/confirm-approve.util.js

```diff
@@ -1,28 +1,27 @@
 import { decimalToHex } from '../../helpers/utils/conversions.util'
 import { calcTokenValue } from '../../helpers/utils/token-util.js'
+import { getTokenData } from '../../helpers/utils/transactions.util'
 
-export function getCustomTxParamsData (data, { customPermissionAmount, tokenAmount, decimals }) {
-  if (customPermissionAmount) {
-    const tokenValue = decimalToHex(calcTokenValue(tokenAmount, decimals))
+export function getCustomTxParamsData (data, { customPermissionAmount, decimals }) {
+  const tokenData = getTokenData(data)
 
-    const re = new RegExp('(^.+)' + tokenValue + '$')
-    const matches = re.exec(data)
-
-    if (!matches || !matches[1]) {
-      return data
-    }
-    let dataWithoutCurrentAmount = matches[1]
-    const customPermissionValue = decimalToHex(calcTokenValue(Number(customPermissionAmount), decimals))
```

Comment:
Is this the erroneous conversion to `Number` mentioned in the PR description?

Reply:
Technically no, it's on line 46 of `confirm-approve.container.js`, and it's still there.

Arguably it's no longer a bug if it's only used for display? 🤔 So I hadn't removed it.

There was a second case of erroneously converting to `Number` as well, which is here.

Reply:
Actually maybe it is a bug - I'll revisit this, to prevent rounding errors in the displayed value.

Reply:
Right. I understand now. The match that was failing was on line 9.

Reply:
Yes, exactly

Reply:
I've removed the problematic line, and am now passing the token limits and amount as strings: https://github.com/MetaMask/metamask-extension/pull/7919/commits/fb39ad75c40a1ec09861245b0c3cf5688e9011e0